File: strings/special.py
import re 

def special(s):
    arr = s.split(' ')
    result = []

    for i in arr:
        temp = re.findall('[a-zA-Z0-9]', i)
        if len(temp) > 0:
            result.append(''.join(temp))

    return " ".join(result)


str1 = "/*Jon is @developer & musician"


print(special(str1))

File: strings/special.js


const special = str => {
    let result = [];
    let arr = str.split(' ');

    for (let x in arr) {
        let temp = arr[x].match(/[a-zA-Z0-9]/g);
        if (temp && temp.length > 0) {
            result.push(temp.join(''));
        }
    }
    return result.join(' ');
}


let str1 = "/*Jon is @developer & musician";

console.log(special(str1));